pkg/apis/core/validation/validation.go

```diff
@@ -3551,11 +3551,28 @@ func ValidatePreferredSchedulingTerms(terms []core.PreferredSchedulingTerm, fldP
 	return allErrs
 }
 
+func validatePodAffinityTermNamespaceSelector(podAffinityTerm core.PodAffinityTerm, fldPath *field.Path) field.ErrorList {
+	nsSelector := podAffinityTerm.NamespaceSelector
+	if nsSelector == nil {
+		return nil
+	}
+
+	allErrs := unversionedvalidation.ValidateLabelSelector(nsSelector, fldPath)
+	for i, m := range nsSelector.MatchExpressions {
+		if m.Operator != metav1.LabelSelectorOpIn {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("matchExpressions").Index(i).Child("operator"), m.Operator, `must be "In"`))
+		}
+	}
+	return allErrs
+}
+
 // validatePodAffinityTerm tests that the specified podAffinityTerm fields have valid data
 func validatePodAffinityTerm(podAffinityTerm core.PodAffinityTerm, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	allErrs = append(allErrs, unversionedvalidation.ValidateLabelSelector(podAffinityTerm.LabelSelector, fldPath.Child("matchExpressions"))...)
+	allErrs = append(allErrs, unversionedvalidation.ValidateLabelSelector(podAffinityTerm.LabelSelector, fldPath.Child("labelSelector"))...)
```

Comment:
Interesting catch... The test used to keep using `fldPath.Child("matchExpressions")`..